include/kokkos-utils/callbacks/TimerListener.hpp

```diff
@@ -0,0 +1,169 @@
+#ifndef KOKKOS_UTILS_CALLBACKS__TIMERLISTENER_HPP
+#define KOKKOS_UTILS_CALLBACKS__TIMERLISTENER_HPP
+
+#include "Kokkos_NumericTraits.hpp"
+
+#include "kokkos-utils/callbacks/Listener.hpp"
+#include "kokkos-utils/callbacks/Matcher.hpp"
+#include "kokkos-utils/concepts/ExecutionSpace.hpp"
+#include "kokkos-utils/timer/Timer.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+
+template <typename...>
+class TimerListener;
+
+/**
+ * @brief Listener that starts a timer when a begin event is received and stops it
+ *        when a corresponding end event is received.
+ *
+ * The start event is selected by requiring it to be of type @p BeginEventType and
+ * to be a match for the passed matcher. The stop event must be of type @p EndEventType
```

Comment:
```suggestion
 * to match @ref matcher. The stop event must be of type @p EndEventType
```

Reply:
I tried but Doxygen gives an error saying that it doesn't know the reference `matcher`. I'm not sure what the issue is, but ll try more to make it work.

Reply:
Outdated.